shared/chat/conversation/info-panel/menu/index.js

```diff
@@ -33,61 +35,70 @@ const Header = ({teamname, memberCount}: {teamname: string, memberCount: number}
   </Kb.Box>
 )
 
-const InfoPanelMenu = (props: Props) => {
-  const addPeopleItems = [
-    {
-      title: 'Add someone by username',
-      subTitle: 'Keybase, Twitter, etc.',
-      onClick: props.onAddPeople,
-      style: {borderTopWidth: 0},
-    },
-    {
-      title: Styles.isMobile ? 'Add someone from address book' : 'Add someone by email',
-      onClick: props.onInvite,
-    },
-  ]
-  const channelItem = props.isSmallTeam
-    ? {
-        onClick: props.onManageChannels,
-        subTitle: props.manageChannelsSubtitle,
-        title: props.manageChannelsTitle,
-      }
-    : {
-        onClick: props.onManageChannels,
-        title: props.manageChannelsTitle,
-        view: (
-          <Kb.Box style={Styles.globalStyles.flexBoxRow}>
-            <Kb.Text style={styles.text} type={Styles.isMobile ? 'BodyBig' : 'Body'}>
-              {props.manageChannelsTitle}
-            </Kb.Text>
-            {props.badgeSubscribe && <Kb.Box style={styles.badge} />}
-          </Kb.Box>
-        ),
-      }
+class InfoPanelMenu extends React.Component<Props> {
+  componentDidDUpdate(prevProps: Props) {
+    if (this.props.hasCanPerform && this.props.visible !== prevProps.visible) {
```

Comment:
What's the reason for checking the diff on visibility?

Reply:
Because by default it's not visible. Only on hover. It's still mounted though. So as you scroll around it'd call this (for no reason). Now it'll only call when you click the menu